src/pages/ResearcherConsole.js

```diff
@@ -152,19 +112,32 @@ export default function ResearcherConsole() {
     }
   };
 
+
+  //Draft delete, by referenceIds
+  const deleteDraftById = async ({ referenceId, collectionsClone }) => {
+    await DAR.deleteDar(referenceId);
+    const targetIndex = findIndex((draft) => {
+      return draft.referenceId === referenceId;
+    })(collectionsClone);
+
+    return targetIndex;
+  };
+
   //Draft delete, passed down to draft table to be used with delete button
-  const deleteDraft = async ({ referenceId, identifier }) => {
+  const deleteDraft = async ({ referenceIds, identifier }) => {
     try {
-      await DAR.deleteDar(referenceId);
       const collectionsClone = cloneDeep(researcherCollections);
-      const targetIndex = findIndex((draft) => {
-        return draft.referenceId === referenceId;
-      })(collectionsClone);
-      if (targetIndex === -1) {
+      const deletedDraftIndices = referenceIds.map((referenceId) => deleteDraftById({ referenceId }));
+      const hasAnySuccess = deletedDraftIndices.findIndex((draftIndex) => draftIndex !== -1);
+
+      if (!hasAnySuccess) {
```

Comment:
When testing locally, the deletion went through, but I kept hitting this error notification. I was able to fix it with the code below, although for drafts it should be sufficient to just use `referenceIds[0]` rather than iterating through the `referenceIds` array
```
  const deleteDraftById = async ({ referenceId, collectionsClone }) => {
    await DAR.deleteDar(referenceId);
    const targetIndex = findIndex((draft) => {
      return contains(draft.referenceIds, referenceId);
    })(collectionsClone);

    return targetIndex;
  };

  //Draft delete, passed down to draft table to be used with delete button
  const deleteDraft = async ({ referenceIds, name }) => {
    try {
      const collectionsClone = cloneDeep(researcherCollections);
      const deletedDraftIndices = await Promise.all(referenceIds.map((referenceId) => deleteDraftById({ referenceId, collectionsClone })));
      const deletionSuccessful = every((draftIndex) => draftIndex !== -1)(deletedDraftIndices);

      if (!deletionSuccessful) {
        Notifications.showError({ text: 'Error processing delete request' });
      } else {
        // if deleted index, remove it from the collections array
        const filteredCollections = collectionsClone.filter((collection, collectionIndex) => {
          return deletedDraftIndices.indexOf(collectionIndex) === -1;
        });
        setResearcherCollections(filteredCollections);
        Notifications.showSuccess({text: `Deleted Data Access Request Draft ${name}`});
      }
    } catch (error) {
      Notifications.showError({
        text: `Failed to delete Data Access Request Draft ${name}`,
      });
    }
  };
```

Reply:
> When testing locally, the deletion went through, but I kept hitting this error notification. I was able to fix it with the code below, although for drafts it should be sufficient to just use `referenceIds[0]` rather than iterating through the `referenceIds` array
> 
> ```
>   const deleteDraftById = async ({ referenceId, collectionsClone }) => {
>     await DAR.deleteDar(referenceId);
>     const targetIndex = findIndex((draft) => {
>       return contains(draft.referenceIds, referenceId);
>     })(collectionsClone);
> 
>     return targetIndex;
>   };
> 
>   //Draft delete, passed down to draft table to be used with delete button
>   const deleteDraft = async ({ referenceIds, name }) => {
>     try {
>       const collectionsClone = cloneDeep(researcherCollections);
>       const deletedDraftIndices = await Promise.all(referenceIds.map((referenceId) => deleteDraftById({ referenceId, collectionsClone })));
>       const deletionSuccessful = every((draftIndex) => draftIndex !== -1)(deletedDraftIndices);
> 
>       if (!deletionSuccessful) {
>         Notifications.showError({ text: 'Error processing delete request' });
>       } else {
>         // if deleted index, remove it from the collections array
>         const filteredCollections = collectionsClone.filter((collection, collectionIndex) => {
>           return deletedDraftIndices.indexOf(collectionIndex) === -1;
>         });
>         setResearcherCollections(filteredCollections);
>         Notifications.showSuccess({text: `Deleted Data Access Request Draft ${name}`});
>       }
>     } catch (error) {
>       Notifications.showError({
>         text: `Failed to delete Data Access Request Draft ${name}`,
>       });
>     }
>   };
> ```

Thank you for this codeblock! It's very clean! If it's sufficient to just use referenceIds[0], i'm going to try going with that solution